WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -854,17 +862,85 @@ private extension AuthenticationManager {
                 // show generic error
                 await MainActor.run {
                     DDLogError("⛔️ Error generating application password: \(error)")
-                    let alert = FancyAlertViewController.makeApplicationPasswordAlert(retryAction: { [weak self] in
-                        self?.checkApplicationPassword(for: siteURL, with: useCase, in: navigationController, onSuccess: onSuccess)
-                    }, restartLoginAction: {
-                        ServiceLocator.stores.deauthenticate()
-                        navigationController.popToRootViewController(animated: true)
-                    })
+                    let alert = FancyAlertViewController.makeSiteCredentialLoginAlert(
+                        message: Localization.applicationPasswordError,
+                        retryAction: { [weak self] in
+                            self?.checkApplicationPassword(for: siteURL, with: useCase, in: navigationController, onSuccess: onSuccess)
+                        },
+                        restartLoginAction: {
+                            ServiceLocator.stores.deauthenticate()
+                            navigationController.popToRootViewController(animated: true)
+                        }
+                    )
+                    navigationController.present(alert, animated: true)
+                }
+            }
+        }
+    }
+
+    /// Checks role eligibility for the logged in user with the site address saved in the credentials.
+    /// Placeholder store ID is used because we are checking for users logging in with site credentials.
+    ///
+    func checkRoleEligibility(in navigationController: UINavigationController, onSuccess: @escaping () -> Void) {
+        roleEligibilityUseCase.checkEligibility(for: WooConstants.placeholderStoreID) { [weak self] result in
+            guard let self else { return }
+            switch result {
+            case .success:
+                onSuccess()
+            case .failure(let error):
+                if case let RoleEligibilityError.insufficientRole(errorInfo) = error {
+                    self.showRoleErrorScreen(for: WooConstants.placeholderStoreID,
+                                             errorInfo: errorInfo,
+                                             in: navigationController,
+                                             onSuccess: onSuccess)
```

Comment:
Unrelated to this PR:
<img src="https://user-images.githubusercontent.com/524475/210298166-4149a21a-05f9-4b6f-ba13-480021964c9e.png" width="300"/>

Tapping `Retry` doesn't show a loading indicator.

Reply:
I added a quick fix in 0cd2149de6a277f947dde9609f2ea2f432f0a300 and 6b078f3c223ef8b2f540b12a683d4818519e6d2e. Here's a recording of the result:

https://user-images.githubusercontent.com/5533851/210319054-03a083c7-92fe-4a0e-b3a2-112354e86d0f.mp4